// @vkontakte/vkui/dist/es6/lib/browser.js
import _slicedToArray from "@babel/runtime/helpers/slicedToArray";
import { IOS, platform } from './platform';
var memoized = {};
export function computeBrowserInfo(userAgent) {
  if (memoized[userAgent]) {
    return memoized[userAgent];
  }

  var platformName = platform(userAgent);
  var platformVersion = null;

  if (platformName === IOS) {
    platformVersion = parseiOSVersion(userAgent);
  }

  var browserInfo = {
    userAgent: userAgent,
    platform: platformName,
    platformVersion: platformVersion
  };
  memoized[userAgent] = browserInfo;
  return browserInfo;
}
export function parseiOSVersion(userAgent) {
  if (!userAgent) {
    return null;
  }

  var match = userAgent.match(/(iphone os|cpu os|ios) \b([0-9]+_[0-9]+(?:_[0-9]+)?)\b/i);

  if (!match) {
    return null;
  }

  var _match$2$replace$spli = match[2].replace(/_/g, '.').split('.'),
      _match$2$replace$spli2 = _slicedToArray(_match$2$replace$spli, 3),
      major = _match$2$replace$spli2[0],
      minor = _match$2$replace$spli2[1],
      patch = _match$2$replace$spli2[2];

  return {
    major: parseInt(major) || 0,
    minor: parseInt(minor) || 0,
    patch: parseInt(patch) || 0
  };
}
