// @vkontakte/vkui/dist/components/ModalCard/ModalCard.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _Button = _interopRequireDefault(require("../Button/Button"));

var _PanelHeaderButton = _interopRequireDefault(require("../PanelHeaderButton/PanelHeaderButton"));

var _getClassName = _interopRequireDefault(require("../../helpers/getClassName"));

var _classNames = _interopRequireDefault(require("../../lib/classNames"));

var _withInsets = _interopRequireDefault(require("../../hoc/withInsets"));

var _dismiss = _interopRequireDefault(require("@vkontakte/icons/dist/24/dismiss"));

var _platform = require("../../lib/platform");

var _utils = require("../../lib/utils");

var _withPlatform = _interopRequireDefault(require("../../hoc/withPlatform"));

var _Subhead = _interopRequireDefault(require("../Typography/Subhead/Subhead"));

var _Title = _interopRequireDefault(require("../Typography/Title/Title"));

var ModalCard = function ModalCard(props) {
  var insets = props.insets,
      icon = props.icon,
      header = props.header,
      caption = props.caption,
      children = props.children,
      actions = props.actions,
      actionsLayout = props.actionsLayout,
      onClose = props.onClose,
      platform = props.platform,
      className = props.className;

  var onButtonClick = function onButtonClick(event) {
    var target = event.currentTarget; // eslint-disable-next-line @typescript-eslint/unbound-method

    var action = actions[Number(target.dataset.index)].action;
    event.persist();

    if (typeof action === 'function') {
      action(event);
    }
  };

  var canShowCloseBtn = platform === _platform.IOS;
  return /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _classNames.default)((0, _getClassName.default)('ModalCard', platform), className)
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "ModalCard__in"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "ModalCard__container",
    style: (0, _utils.isNumeric)(insets.bottom) ? {
      marginBottom: insets.bottom
    } : null
  }, (0, _utils.hasReactNode)(icon) && /*#__PURE__*/_react.default.createElement("div", {
    className: "ModalCard__icon"
  }, icon), (0, _utils.hasReactNode)(header) && /*#__PURE__*/_react.default.createElement(_Title.default, {
    level: "2",
    weight: "semibold",
    className: "ModalCard__header"
  }, header), (0, _utils.hasReactNode)(caption) && /*#__PURE__*/_react.default.createElement(_Subhead.default, {
    weight: "regular",
    className: "ModalCard__subheader"
  }, caption), children, actions.length > 0 && /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _classNames.default)('ModalCard__actions', {
      'ModalCard__actions--v': actionsLayout === 'vertical'
    })
  }, actions.map(function (_ref, i) {
    var title = _ref.title,
        mode = _ref.mode;
    return /*#__PURE__*/_react.default.createElement(_Button.default, {
      key: i,
      "data-index": i,
      size: "xl",
      mode: mode,
      onClick: onButtonClick
    }, title);
  })), canShowCloseBtn && /*#__PURE__*/_react.default.createElement(_PanelHeaderButton.default, {
    className: "ModalCard__dismiss",
    onClick: onClose
  }, /*#__PURE__*/_react.default.createElement(_dismiss.default, null)))));
};

ModalCard.defaultProps = {
  actions: [],
  actionsLayout: 'horizontal',
  insets: {}
};

var _default = (0, _withPlatform.default)((0, _withInsets.default)(ModalCard));

exports.default = _default;
