// @vkontakte/vkui/dist/lib/browser.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.computeBrowserInfo = computeBrowserInfo;
exports.parseiOSVersion = parseiOSVersion;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _platform = require("./platform");

var memoized = {};

function computeBrowserInfo(userAgent) {
  if (memoized[userAgent]) {
    return memoized[userAgent];
  }

  var platformName = (0, _platform.platform)(userAgent);
  var platformVersion = null;

  if (platformName === _platform.IOS) {
    platformVersion = parseiOSVersion(userAgent);
  }

  var browserInfo = {
    userAgent: userAgent,
    platform: platformName,
    platformVersion: platformVersion
  };
  memoized[userAgent] = browserInfo;
  return browserInfo;
}

function parseiOSVersion(userAgent) {
  if (!userAgent) {
    return null;
  }

  var match = userAgent.match(/(iphone os|cpu os|ios) \b([0-9]+_[0-9]+(?:_[0-9]+)?)\b/i);

  if (!match) {
    return null;
  }

  var _match$2$replace$spli = match[2].replace(/_/g, '.').split('.'),
      _match$2$replace$spli2 = (0, _slicedToArray2.default)(_match$2$replace$spli, 3),
      major = _match$2$replace$spli2[0],
      minor = _match$2$replace$spli2[1],
      patch = _match$2$replace$spli2[2];

  return {
    major: parseInt(major) || 0,
    minor: parseInt(minor) || 0,
    patch: parseInt(patch) || 0
  };
}
